### PR TITLE
Refactor Separate the coupled GPU task functions.

### PR DESCRIPTION
**Pull request by A-006**
### What's changed?
In rho, force, and vlocal, these functions have many parameters and do not adhere to the principle of a function performing a single task. They should be separated.

### CODE REVIEW

File: source/module_hamilt_lcao/module_gint/gint_force_gpu.cu [hunk 1]
```diff
@@ -248,6 +265,35 @@ void gint_fvl_gamma_gpu(hamilt::HContainer<double>* dm,
     {
         checkCuda(cudaStreamSynchronize(gridt.streams[i]));
     }
+    double stress_sum[6 *gridt.nstreams] = {0.0};
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/gint_force_gpu.cu]: you can not define array like this

File: source/module_hamilt_lcao/module_gint/gint_force_gpu.cu [hunk 2]
```diff
@@ -203,51 +214,76 @@ void gint_fvl_gamma_gpu(hamilt::HContainer<double>* dm,
...
+                            sizeof(double) * 6 * gridt.nstreams,
+                            cudaMemcpyDeviceToHost));
+        for (int i = 0; i < 6; i++)
+        {
+            for (int j = 0; j < gridt.nstreams; j++)
+            {
+                stress[i] += stress_host[j * 6 + i];
+            }
+        }
+        stress_host.clear();
```
Comment (dzzz2001) [inline source/module_hamilt_lcao/module_gint/gint_force_gpu.cu]: stress_host.clear() is redundant here, the same as force_host.clear() below.

File: source/module_hamilt_lcao/module_gint/gtask_force.cpp [hunk 3]
```diff
@@ -211,16 +213,16 @@ void gpu_task_generator_force(const Grid_Technique& gridt,
         }
     }
     atom_pair_num = tid;
-
-    delete[] gpu_mat_cal_flag;
+    gpu_mat_cal_flag.clear();
```
Comment (dzzz2001) [inline source/module_hamilt_lcao/module_gint/gtask_force.cpp:216]: you should remove all std::vector.clear() functions

File: source/module_hamilt_lcao/module_gint/gint_force_gpu.cu (hunk 2 above)
Comment (A-006) [reply source/module_hamilt_lcao/module_gint/gint_force_gpu.cu]: It has been resolved

File: source/module_hamilt_lcao/module_gint/gtask_force.cpp (hunk 3 above)
Comment (A-006) [reply source/module_hamilt_lcao/module_gint/gtask_force.cpp:216]: It has been resolved

**Review by mohanchen — APPROVED** [review]

**Pull request merged**